### PR TITLE
use package:lints; rev pubspec version

### PR DESCRIPTION
**Pull request by devoncarew**
- rev the pubspec version to stable in preparation for publishing
- switch to using package:lints (and address the analysis issues found)
- update the depedabot configuration; update the github action versions used
- add markdown badges to the readme

After this lands I'd publish a new version of the package (which would capture the recent changes; `handle expires_in when encoded as string` and `populate the pubspec `repository` field`).

### CODE REVIEW

**Comment by kevmoo** [comment]
Want to bump min SDK and use pkg:lints, too?

File: analysis_options.yaml [hunk 1]
```diff
@@ -1 +1,40 @@
-include: package:pedantic/analysis_options.yaml
+include: package:lints/recommended.yaml
```
Comment (kevmoo) [inline analysis_options.yaml:1]: need to change pubspec.yaml?

**Comment by devoncarew** [comment]
> Want to bump min SDK and use pkg:lints, too?

Done - bumped to using package:lints 2.0.0.

File: analysis_options.yaml (hunk 1 above)
Comment (devoncarew) [reply analysis_options.yaml:1]: Added a dep on package:lints.

**Comment by kevmoo** [comment]
Need to fix the super param lint!

**Review by natebosch — APPROVED** [review]

**Pull request merged**